Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -221,9 +221,29 @@ class PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
+    func purchase(sk1Product: SK1Product,
```

Comment:
Yeah not sure about this API. It's gonna become even more convoluted with #1160 once the package can be 2 different types of product. Perhaps we should combine both PRs because they'll need to work together.